jetty-websocket/javax-websocket-client-impl/src/main/java/org/eclipse/jetty/websocket/jsr356/ClientContainer.java

```diff
@@ -482,10 +482,14 @@ public void setAsyncSendTimeout(long ms)
     @Override
     public void setDefaultMaxBinaryMessageBufferSize(int max)
     {
-        // overall message limit (used in non-streaming)
-        client.getPolicy().setMaxBinaryMessageSize(max);
         // incoming streaming buffer size
         client.setMaxBinaryMessageBufferSize(max);
+
+        // bump overall message limit (used in non-streaming) if buffer max is larger
+        if(client.getPolicy().getMaxBinaryMessageSize() < max)
```

Comment:
I don't think this is correct. You should unconditionally update the `maxMessageSize` as it was before. Simple setters that only work in certain conditions are always a surprise.